nflow-engine/src/main/java/io/nflow/engine/workflow/curated/MaintenanceWorkflow.java

```diff
@@ -0,0 +1,52 @@
+package io.nflow.engine.workflow.curated;
+
+import io.nflow.engine.service.MaintenanceConfiguration;
+import io.nflow.engine.service.MaintenanceResults;
+import io.nflow.engine.service.MaintenanceService;
+import io.nflow.engine.workflow.definition.NextAction;
+import io.nflow.engine.workflow.definition.StateExecution;
+import io.nflow.engine.workflow.definition.StateVar;
+import org.springframework.stereotype.Component;
+
+import javax.inject.Inject;
+
+import static io.nflow.engine.workflow.curated.CronWorkflow.State.schedule;
+import static io.nflow.engine.workflow.definition.NextAction.moveToState;
+
+/**
+ * Workflow that cleans up workflow instances periodically.
+ */
+@Component
+public class MaintenanceWorkflow extends CronWorkflow {
+  public static final String MAINTENANCE_WORKFLOW_TYPE = "nFlowMaintenance";
+  public static final String VAR_MAINTENANCE_CONFIGURATION = "config";
+
+  @Inject
+  MaintenanceService maintenanceService;
+
+  public MaintenanceWorkflow() {
+    super(MAINTENANCE_WORKFLOW_TYPE);
```

Comment:
It should be possible to override workflow settings of MaintenanceWorkflow

Reply:
I see no reason for anyone to ever extend MaintenanceWorkflow. It is much more cleaner to just extend CronWorkflow directly

Reply:
Overriding the handleFailure for one. Overriding the workflow settings for two. Everything else in MaintenanceWorkflow is reusable, I would not want to re-implement those parts. I tried, extending MaintenanceWorkflow was much more cleaner.